src/std/misc/atom.ss

```diff
@@ -108,11 +109,10 @@
 
 (def (atomic-counter (initial-value -1))
   (let (a (atom initial-value))
-    (lambda ()
+    (lambda ((increment 1))
```

Comment:
please dont.

This is an internal lambda and internal opt-lambdas have no direct dispatch. So we have to pack the arguments, and dispatch on arity with apply.

This is unreasonably slow.

Reply:
So two options here: either always increment by one or always pass the increment; no middle ground.

Reply:
Would a case-lambda do?

Reply:
I yield.